docs/reference/query-dsl/percolator-query.asciidoc

```diff
@@ -0,0 +1,424 @@
+    "my-type": {
+      "properties": {
+        "message": {
+          "type": "string"
+        }
+      }
+    }
+  }
+}'
+--------------------------------------------------
+
+Register a query in the percolator:
+
+[source,js]
+--------------------------------------------------
+curl -XPUT 'localhost:9200/my-index/.percolator/1' -d '{
+    "query" : {
+        "match" : {
+            "message" : "bonsai tree"
+        }
+    }
+}'
+--------------------------------------------------
+
+Match a document to the registered percolator queries:
+
+[source,js]
+--------------------------------------------------
+curl -XGET 'localhost:9200/my-index/_search' -d '{
+    "query" : {
+        "percolator" : {
+            "document_type" : "my-type",
+            "document" : {
+                "message" : "A new bonsai tree in the office"
+            }
+        }
+    }
+}'
+--------------------------------------------------
+
+The above request will yield the following response:
+
+[source,js]
+--------------------------------------------------
+{
+  "took": 5,
+  "timed_out": false,
+  "_shards": {
+    "total": 5,
+    "successful": 5,
+    "failed": 0
+  },
+  "hits": {
+    "total": 1,
+    "max_score": 0,
+    "hits": [
+      { <1>
+        "_index": "my-index",
+        "_type": ".percolator",
+        "_id": "1",
+        "_score": 0,
+        "_source": {
+          "query": {
+            "match": {
+              "message": "bonsai tree"
+            }
+          }
+        }
+      }
+    ]
+  }
+}
+--------------------------------------------------
+
+<1> The percolate query with id `1` matches our document.
+
+[float]
+=== Indexing Percolator Queries
+
+Percolate queries are stored as documents in a specific format and in an arbitrary index under a reserved type with the
+name `.percolator`. The query itself is placed as is in a JSON object under the top level field `query`.
+
+[source,js]
+--------------------------------------------------
+{
+    "query" : {
+		"match" : {
+			"field" : "value"
+		}
+	}
+}
+--------------------------------------------------
+
+Since this is just an ordinary document, any field can be added to this document. This can be useful later on to only
+percolate documents by specific queries.
+
+[source,js]
+--------------------------------------------------
+{
+	"query" : {
+		"match" : {
+			"field" : "value"
+		}
+	},
+	"priority" : "high"
+}
+--------------------------------------------------
+
+Just as with any other type, the `.percolator` type has a mapping, which you can configure via the mappings APIs.
+The default percolate mapping doesn't index the query field, only stores it.
+
+Because `.percolate` is a type it also has a mapping. By default the following mapping is active:
+
+[source,js]
+--------------------------------------------------
+{
+	".percolator" : {
+		"properties" : {
+			"query" : {
+				"type" : "percolator"
+			}
+		}
+	}
+}
+--------------------------------------------------
```

Comment:
can you add a few lines about the percolator field under `docs/reference/mapping/fields/`?